lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -1200,4 +1200,65 @@ class ProviderTest {
 
         assertNotNull(response)
     }
+
+    @Test
+    fun `batch call contract with block hash and block tag`() {
+        val call1 = Call(
+            contractAddress = balanceContractAddress,
+            entrypoint = "get_balance",
+            calldata = emptyList(),
+        )
+        val blockHash = provider.getBlockHashAndNumber().send().blockHash
+
+        val call2 = Call(
+            contractAddress = balanceContractAddress,
+            entrypoint = "get_balance",
+            calldata = emptyList(),
+        )
+
+        val providerCalls = listOf(
+            provider.callContract(
+                call = call1,
+                blockHash = blockHash,
+            ),
+            provider.callContract(
+                call = call2,
+                blockTag = BlockTag.LATEST,
+            ),
+        )
+        val request = provider.batchRequests(providerCalls)
+        val response = request.send()
+        val expectedBalance = provider.getStorageAt(balanceContractAddress, selectorFromName("balance"), BlockTag.LATEST).send()
+
+        assertEquals(response[0].first(), expectedBalance)
+        assertEquals(response[1].first(), expectedBalance)
+    }
+
+    @Test
+    fun `batch get transactions by block number and index`() {
+        val blockNumber = provider.getBlockNumber().send()
+        val request = provider.batchRequests(
+            provider.getTransactionByBlockIdAndIndex(blockNumber, 0),
+            provider.getTransactionByBlockIdAndIndex(blockNumber, 0),
+        )
+
+        val response = request.send()
+
+        assertNotNull(response[0])
+        assertNotNull(response[1])
+    }
+
+    @Test
+    fun `batch get transactions with polymorphic serializers`() {
+        val blockNumber = provider.getBlockNumber().send()
+        val request = provider.batchRequests(
+            provider.getTransactionByBlockIdAndIndex(blockNumber, 0),
+            provider.getTransaction(invokeTransactionHash),
```

Comment:
We can probably delete `batch get transactions by block number and index` since it can be the part of this test.